flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/aggregate/DataSetSlideWindowAggReduceCombineFunction.scala

```diff
@@ -52,19 +55,35 @@ class DataSetSlideWindowAggReduceCombineFunction(
 
   private val intermediateRow: Row = new Row(keysAndAggregatesArity + 1)
 
+  protected var preAggfunction: GeneratedAggregations = _
+
+  override def open(config: Configuration): Unit = {
```

Comment:
This function requires two code-gen'd functions because it handles partial and final aggregation. One function is for the preaggregation, the other for the final aggregation.
`DataSetSlideWindowAggReduceGroupFunction` can also be use stand-alone for the non-combinable case but also as final aggregation in the combinable case. So the code-gen function for the final aggregation is compiled and instanciated by `DataSetSlideWindowAggReduceGroupFunction` and the partial code-gen'd function is taken care of by `DataSetSlideWindowAggReduceCombineFunction`.

Does that make sense?